Main.py

```diff
@@ -126,8 +126,9 @@ def generate(settings, window=dummy_window()):
 def build_world_graphs(settings, window=dummy_window()):
     logger = logging.getLogger('')
     worlds = []
-    for i in range(0, settings.world_count):
-        worlds.append(World(i, settings))
+    for i in range(1, settings.world_count):
+        worlds.append(World(i, copy.copy(settings)))
+    worlds.insert(0, World(0, settings))
```

Comment:
World0 has to be added last since the World constructor changes things on the Settings object, so if the first one is copied by reference then modified, then the modified version will be copied to all the other worlds as well. Though that probably would have been good to note in a comment.

Reply:
Hm, I see. Maybe we should just be using copies throughout?

Reply:
The spoiler is generated using the original settings objects so modified settings wouldn't be reflected in the spoiler, and there are possibly other places I'm not aware of that would cause issues

Reply:
Ah, okay got it.